animation.go

```diff
@@ -53,16 +64,50 @@ func NewAnimation(d time.Duration, fn func(float32)) *Animation {
 	return &Animation{Duration: d, Tick: fn}
 }
 
+// NewIndefiniteAnimation creates an indefinite animation where the callback function will be called
+// for every rendered frame once started, until stopped.
+//
+// Since: 2.6
+func NewIndefiniteAnimation(fn func()) *IndefiniteAnimation {
+	return &IndefiniteAnimation{Tick: fn}
+}
+
 // Start registers the animation with the application run-loop and starts its execution.
 func (a *Animation) Start() {
 	CurrentApp().Driver().StartAnimation(a)
 }
 
+// Start registers the animation with the application run-loop and starts its execution.
+func (i *IndefiniteAnimation) Start() {
+	i.setupAnimation()
+	i.animation.Start()
+}
+
 // Stop will end this animation and remove it from the run-loop.
 func (a *Animation) Stop() {
 	CurrentApp().Driver().StopAnimation(a)
 }
 
+// Stop will end this animation and remove it from the run-loop.
+func (i *IndefiniteAnimation) Stop() {
+	i.setupAnimation()
+	i.animation.Stop()
+}
+
+func (i *IndefiniteAnimation) setupAnimation() {
+	if !i.isSetup {
+		i.animation = Animation{
+			Tick: func(_ float32) {
+				i.Tick()
+			},
+			Curve:       AnimationLinear, // any curve will work
+			Duration:    1 * time.Second, // anything positive here will work
+			RepeatCount: AnimationRepeatForever,
+		}
+		i.isSetup = true
+	}
```

Comment:
It would match our style to indent less by instead inverting your if-statement and retuning if it is set up.